protobuf-serde/src/lib.rs

```diff
@@ -0,0 +1,24 @@
+extern crate serde;
+extern crate protobuf;
+
+use protobuf::SingularPtrField;
+
+use serde::Serialize;
+use serde::Deserialize;
+
+use serde::ser::Serializer;
+use serde::Deserializer;
+
+pub fn serialize_singular_ptr_field<'a, T: Serialize, S>(spf: &SingularPtrField<T>, serializer: S)
+                                                         -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
+    where S: Serializer {
+    spf.get_ref().serialize(serializer)
+}
+
+pub fn deserialize_singular_ptr_field<'de, D, T>(deserializer: D)
+                                                 -> Result<SingularPtrField<T>, D::Error>
+    where D: Deserializer<'de>, T: Deserialize<'de>
+{
+    Ok(SingularPtrField::some(Option::deserialize(deserializer).unwrap().unwrap()))
```

Comment:
This unwraps must be mistakes.

If it is there temporarlily, please add TODO.

Reply:
If unwrap is the right thing, it deserved an explanation in comments at least.

Reply:
Ah ok I understand now - adding another test as we speak.

Reply:
> Ah ok I understand now - adding another test as we speak.

No need to add all possible tests right now. I'm just noticed `unwrap` so I'm not convinced it's safe, so I'm saying that either:
* code should be rewritten without unwrap
* or code should have comment explanations why the panic is not possible